htdocs/core/lib/functions.lib.php

```diff
@@ -7483,7 +7483,7 @@ function dol_textishtml($msg, $option = 0)
 			return true;
 		} elseif (preg_match('/<\/textarea/i', $msg)) {
 			return true;
-		} elseif (preg_match('/<(b|em|i|u)>/i', $msg)) {
+		} elseif (preg_match('/<(b|em|i|u)[^>]*>/i', $msg)) {
```

Comment:
May be we can protect better with
preg_match('/<(b|em|i|u)(\s*[^>]+)?>/i', $msg)
so nothing after b em i u 
or   a space and something no > until the >
Can you check this ?
Also, you should be able to complete the phpunit tests cases with an example that was not correctly detected and he is after.

Reply:
You're right !
Is this test ok ?